lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioInputStream.java

```diff
@@ -115,7 +115,7 @@ public int read(byte[] bytes, int offset, int length)
 
         checkFromIndexSize(offset, length, bytes.length);
         if (length == 0) {
```

Comment:
@jkylling  FYI

Reply:
I reverted this, it was an issue with tests, not Alluxio. Sorry about the false alarm.

Reply:
We might want to do some fuzzing of the various file system implementations. They have inconsistent behaviour at edge cases of file legnth, offset and buffer length. Sometimes one implementation will throw an exception, while the other will return 0 bytes read. The fuzzing in https://github.com/trinodb/trino/blob/2188fde6c4c1cd9f4db46f4745322b19b184341b/lib/trino-filesystem-cache-alluxio/src/test/java/io/trino/filesystem/alluxio/TestFuzzAlluxioCacheFileSystem.java#L88-L89 caught one issue like this, but I believe there are more.

Reply:
I just found we're affected by https://github.com/aws/aws-sdk-java-v2/issues/3538